packages/python-packages/api-stub-generator/apistub/_apiview.py

```diff
@@ -120,27 +127,36 @@ def add_type(self, type_name, id=None):
         # Those types needs to be recursively processed
         multi_types = re.search(COMPLEX_DATA_TYPE_PATTERN,type_name)
         if multi_types:
-            self._generate_type_tokens(type_name, multi_types.groups()[0])
+            self._generate_type_tokens(type_name, multi_types.groups()[0], line_id)
         else:
             # Encode mutliple types with or seperator into Union
             types = [t for t in type_name.split() if t != 'or']
             if len(types) > 1:
                 # Make a Union of types if multiple types are present
-                self._generate_type_tokens("Union[{}]".format(", ".join(types)), "Union")
+                self._generate_type_tokens("Union[{}]".format(", ".join(types)), "Union", line_id)
             else:
-                self._add_type_token(types[0])
+                self._add_type_token(types[0], line_id)
 
 
-    def _add_type_token(self, type_name):
+    def _add_type_token(self, type_name, line_id = None):
         token = Token(type_name, TokenKind.TypeName)
         type_full_name = type_name[1:] if type_name.startswith("~") else type_name
         token.set_value(type_full_name.split(".")[-1])
         navigate_to_id = self.nodeindex.get_id(type_full_name)
         if navigate_to_id:
             token.NavigateToId = navigate_to_id
+        elif type_name.startswith("~") and line_id:
+            # if navigation ID is missing for internal type, add diagnostic error
+            self.add_diagnostic(SOURCE_LINK_NOT_AVAILABLE.format(token.Value), line_id)            
         self.add_token(token)
 
 
+    def add_diagnostic(self, text, line_id):
+        self.Diagnostics.append(Diagnostic(line_id, text))
+        # log diagnostic as error
+        logging.error(text)
```

Comment:
Is the line ID useful information to include in the text?

Reply:
@annatisch: It's for API tool to tag which line this diagnostic error is created for. I have removed this log line and created a better report with class and method name reference. Line Id is more of an internal ID created for tool and will be confusing to dev.